fsx/main.tf

```diff
@@ -0,0 +1,327 @@
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+  */
+  egress {
+    from_port   = 0
+    to_port     = 0
+    protocol    = "-1"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+}
+
+/**************************************************************************************************
+ * FSxL
+ *************************************************************************************************/
+
+resource "aws_fsx_lustre_file_system" "lustre" {
+  subnet_ids                    = [aws_subnet.public.id]
+  security_group_ids            = [aws_security_group.all.id]
+  deployment_type               = "SCRATCH_2"
+  storage_capacity              = var.lustre_GiB
+  weekly_maintenance_start_time = var.lustre_weekly_maintenance_start_time
+
+  lifecycle {
+    prevent_destroy = true
+  }
+}
+
+/**************************************************************************************************
+ * Batch
+ *************************************************************************************************/
+
+data "cloudinit_config" "all" {
+  # user data scripts for Batch worker instances
+  gzip = false
+
+  part {
+    content_type = "text/x-shellscript"
+    content      = file("${path.module}/../assets/init_docker_instance_storage.sh")
+  }
+
+  part {
+    content_type = "text/x-shellscript"
+    content      = <<-EOT
+    #!/bin/bash
+    set -euxo pipefail
+    # enable EC2 Instance Connect for troubleshooting (if security group allows inbound SSH)
+    yum install -y ec2-instance-connect && grep eic_run_authorized_keys /etc/ssh/sshd_config
+    # mount FSxL
+    amazon-linux-extras install -y lustre2.10
+    mkdir -p /mnt/fsx
+    mount -t lustre -o noatime,flock ${aws_fsx_lustre_file_system.lustre.dns_name}@tcp:/${aws_fsx_lustre_file_system.lustre.mount_name} /mnt/fsx
+    lfs setstripe -E 1G -c 1 -E 16G -c 4 -S 16M -E -1 -c -1 -S 256M /mnt/fsx
+    df -h
+    # Somehow the preceding steps nondeterministically interfere with ECS agent startup. Set a cron
+    # job to keep trying to start it. (We can't simply `systemctl start ecs` here, because the ecs
+    # systemd service requires cloud-init to have finished.)
+    echo "* * * * * root /usr/bin/systemctl start ecs" > /etc/cron.d/ecs-workaround
+    /usr/bin/systemctl reload crond
+    EOT
+  }
+}
+
+resource "aws_iam_instance_profile" "task" {
+  name = "${var.environment_tag}-task"
+  role = aws_iam_role.task.name
+}
+
+resource "aws_launch_template" "task" {
+  name                   = "${var.environment_tag}-task"
+  update_default_version = true
+  iam_instance_profile {
+    name = aws_iam_instance_profile.task.name
+  }
+  user_data = data.cloudinit_config.all.rendered
+}
+
+resource "aws_batch_compute_environment" "task" {
+  compute_environment_name_prefix = "${var.environment_tag}-task"
+  type                            = "MANAGED"
+  service_role                    = aws_iam_role.batch.arn
+
+  compute_resources {
+    type                = "SPOT"
+    instance_type       = ["m5d", "c5d", "r5d"]
```

Comment:
What is the purpose of selecting `d` type instances? Are you making use of the local disk at all?

Reply:
Yea, we [use](https://github.com/miniwdl-ext/miniwdl-aws-terraform/blob/main/assets/init_docker_instance_storage.sh) them for docker storage & scratch, a different tradeoff from EBS autoscaling -- way faster but fixed size